diesel/src/mysql/types/date_and_time.rs

```diff
@@ -78,16 +78,19 @@ impl ToSql<Timestamp, Mysql> for NaiveDateTime {
 impl FromSql<Timestamp, Mysql> for NaiveDateTime {
     fn from_sql(bytes: Option<&[u8]>) -> Result<Self, Box<Error+Send+Sync>> {
         let mysql_time = <ffi::MYSQL_TIME as FromSql<Timestamp, Mysql>>::from_sql(bytes)?;
-        Ok(NaiveDate::from_ymd(
+
+        // Clippy complains if you use `.into()` after `.ok_or()`
```

Comment:
Clippy complains here because this calls a function even in the success case. Creating a `String` from a `&'static str` will allocate memory.

A better solution would be to replace the `ok_or` call with [`ok_or_else`](https://doc.rust-lang.org/stable/std/option/enum.Option.html#method.ok_or_else) and construct the error message string there.

Reply:
Thanks for pointing this out!